examples/parallel-reconstruct.py

```diff
@@ -0,0 +1,170 @@
+from recOrder.compute.reconstructions import (
+    initialize_reconstructor,
+    reconstruct_qlipp_stokes,
+    reconstruct_qlipp_birefringence,
+    reconstruct_phase3D,
+)
+
+# This example demonstrates a 3D-pol-to-birfringence reconstruction applied
+# to each position and time point of a micromanager dataset. The example
+# uses multiprocessing to apply the reconstructions in parallel.
+
+# This example will download a ~50 MB test dataset to your working directory.
+
+N_processes = 6  # (should have at least this many cores available)
+
+
+def precomputation():
+
+    # download test data
+    data_folder = os.path.join(os.getcwd(), "data_temp")
+    os.makedirs(data_folder, exist_ok=True)
+    url = "https://zenodo.org/record/6983916/files/recOrder_test_data.zip?download=1"
+    zip_file = "recOrder_test_Data.zip"
+    output = os.path.join(data_folder, zip_file)
+    if not os.path.exists(output):
+        print("Downloading test files...")
+        download(url, out=output)
+        shutil.unpack_archive(output, extract_dir=data_folder)
+
+    temp_path = os.path.join(data_folder, "temp.zarr")
+    converter = TIFFConverter(
+        os.path.join(
+            data_folder,
+            "2022_08_04_recOrder_pytest_20x_04NA/2T_3P_16Z_128Y_256X_Kazansky_1",
+        ),
+        temp_path,
+    )
+    converter.run()
+
+    # setup input reader
+    reader = open_ome_zarr(temp_path, mode="r+")
+    T, C, Z, Y, X = reader["0/0/0"].data.shape
+    dtype = reader["0/0/0"].data.dtype
+    P = len(list(reader.positions()))
+    bg_data = load_bg(
+        os.path.join(data_folder, "2022_08_04_recOrder_pytest_20x_04NA/BG/"),
+        height=Y,
+        width=X,
+    )
+
+    # setup output zarr
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    writer = open_ome_zarr(
+        "./output/reconstructions_" + timestamp + ".zarr",
+        layout="hcs",
+        mode="w-",
+        channel_names=[
+            "Retardance",
+            "Orientation",
+            "BF - computed",
+            "DoP",
+            "Phase",
+        ],
+    )
+    for p in range(P):
+        position = writer.create_position("0", str(p), "0")
+        position.create_zeros(name="0", shape=(T, 5, Z, Y, X), dtype=dtype)
+
+    return (reader, bg_data, writer)
+
+
+# define the work to be done on a single process
+# in this example a single process will loop through a set of CZYX volumes, and
+# for each volume it will read the data, apply a reconstruction, then save the
+# result to a hcs-format zarr
+def single_process(i, reader, bg_data, writer, vol_start, vol_end):
+    T, C, Z, Y, X = reader["0/0/0"].data.shape
+    for vol in range(vol_start, vol_end + 1):
+        print("DEBUG:", i, vol)
+        p = int(np.floor(vol / T))
+        t = int(vol % T)
+        print(f"Reconstructing vol={vol}, pos={p}, time={t}")
+
+        data = reader["0/" + str(p) + "/0"]["0"][t, ...]
+
+        # Set up a reconstructor.
+        reconstructor_args = {
```

Comment:
I usually create the reconstructor outside of the process since all the reconstructions share the same parameters. I also do the bg_stokes reconstruction outside, because it's the same case.

Reply:
Nope I 100% agree that that's the correct thing to do. Fixed!